### PR TITLE
WIP Method parameters are now correctly replaced in subqueries in projections

### LINKED ISSUE #4254: Method parameters don't get replaced correctly in visitor during projection
### Source/destination types

```csharp
public class Entity{
	public int Id { get; set; }
	
	public int IntToStringRegular { get; set; }
	
	// int IntToStringShadow property on DB
	
	// string MyStringShadow property on DB
}

public class EntityDto{
	public int Id { get; set; }
	
	public string IntToStringRegular { get; set; }
	
	public string IntToStringShadow { get; set; }
	
	public string MyStringShadow { get; set; }
}
```

### Mapping configuration

```csharp
var config = new MapperConfiguration(c =>
{
	c.CreateMap<int, string>()
		.ConvertUsing(i => MyConversionFunction(i));
	
	c.CreateMap<Entity, EntityDto>()
		.ForMember(e => e.IntToStringShadow, c => c.MapFrom(e => EF.Property<int>(e, "IntToStringShadow")))
		.ForMember(e => e.MyStringShadow, c => c.MapFrom(e => EF.Property<string>(e, "MyStringShadow")));
});
```

### Version: 12.0.1

EF Core: 6.0.15

### Expected behavior

The expression should map correctly, something like:
```csharp
DbSet<Entity>()
    .Select(dtoEntity => new EntityDto{ 
        Id = dtoEntity.Id, 
        IntToStringRegular = Test.MyConversionFunction(dtoEntity.IntToStringRegular), 
        IntToStringShadow = Test.MyConversionFunction(EF.Property<int>(dtoEntity, "IntToStringShadow")), 
        MyStringShadow = EF.Property<string>(dtoEntity, "MyStringShadow") 
    }
    )
```

Or even something like this could work I guess, if it's simpler inside the actual logic:
```csharp
DbSet<Entity>()
    .Select(dtoEntity => new Object_111950984___IntToStringShadow_Id_IntToStringRegular{ 
        IntToStringRegular = dtoEntity.IntToStringRegular, 
        Id = dtoEntity.Id, 
        __IntToStringShadow = EF.Property<int>(dtoEntity, "IntToStringShadow"),
        MyStringShadow = EF.Property<string>(dtoEntity, "MyStringShadow") 
    }
    )
    .Select(dtoLet => new EntityDto{ 
        Id = dtoLet.Id, 
        IntToStringRegular = Test.MyConversionFunction(dtoLet.IntToStringRegular), 
        IntToStringShadow = Test.MyConversionFunction(dtoLet.__IntToStringShadow), 
        MyStringShadow = dtoLet.MyStringShadow
    }
    )
```

### Actual behavior

This is what I get:

```csharp
DbSet<Entity>()
    .Select(dtoEntity => new Object_111950984___IntToStringShadow_Id_IntToStringRegular{ 
        IntToStringRegular = dtoEntity.IntToStringRegular, 
        Id = dtoEntity.Id, 
        __IntToStringShadow = EF.Property<int>(dtoEntity, "IntToStringShadow") 
    }
    )
    .Select(dtoLet => new EntityDto{ 
        Id = dtoLet.Id, 
        IntToStringRegular = Test.MyConversionFunction(dtoLet.IntToStringRegular), 
        IntToStringShadow = Test.MyConversionFunction(dtoLet.__IntToStringShadow), 
        MyStringShadow = EF.Property<string>(dtoEntity, "MyStringShadow") 
    }
    )
```

As you can see `MyStringShadow` has the wrong parameter, or it's in the wrong place itself.

### Steps to reproduce
[Fiddle](https://dotnetfiddle.net/msq7Wl)

### PR DESCRIPTION
**Pull request by Xriuk**
Fixes #4254

### CODE REVIEW

**Comment by Xriuk** [comment]
@dotnet-policy-service agree

File: src/UnitTests/Projection/ProjectSubqueryWithParameterAsMethodArgument.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AutoMapper.UnitTests.Projection;
+public class ProjectSubqueryWithParameterAsMethodArgument
+{
+    private MapperConfiguration _config;
+
+    public static string MyConversionFunction(int value) {
```
Comment (lbargaoanu) [inline src/UnitTests/Projection/ProjectSubqueryWithParameterAsMethodArgument.cs:12]: This is client evaluation. We don't support that with `ProjectTo`.

Comment (Xriuk) [reply src/UnitTests/Projection/ProjectSubqueryWithParameterAsMethodArgument.cs:12]: Client evaluation after projection, I thought you supported everything a regular Select does.
Something like:
```
DbSet<MyEntity>().Select(e => new MyEntityDto{
    // ...
    MyField = CustomFunc(e.MyField),
    // ...
});
```
This is a valid Queryable

Comment (lbargaoanu) [reply src/UnitTests/Projection/ProjectSubqueryWithParameterAsMethodArgument.cs:12]: No, `ProjectTo` is a way to write SQL queries.

**Pull request closed without merging**